### PR TITLE
Water balance and sideview fixes

### PR DESCRIPTION
**Pull request by ldebek**

### CODE REVIEW

File: tool_water_balance/tools/waterbalance.py [hunk 1]
```diff
@@ -353,14 +353,16 @@ def get_incoming_and_outcoming_link_ids(self, wb_polygon, model_part):
...
+                pump_end_node_id = pump["node_idx2"]
+                linestring = pump.geometry().asPolyline()
                 # check if flow is in or out by testing if startpoint
                 # is inside polygon --> out
-                outgoing = wb_polygon.contains(QgsPointXY(geom[0]))
+                outgoing = wb_polygon.contains(QgsPointXY(linestring[0]))
                 # check if flow is in or out by testing if endpoint
                 # is inside polygon --> in
-                incoming = wb_polygon.contains(QgsPointXY(geom[-1]))
+                incoming = wb_polygon.contains(QgsPointXY(linestring[-1])) if not pump_end_node_id == -9999 else False
```
Comment (benvanbasten-ns) [inline tool_water_balance/tools/waterbalance.py]: Perhaps define -9999 as constant variable?

File: tool_water_balance/tools/waterbalance.py [hunk 2]
```diff
@@ -353,14 +353,16 @@ def get_incoming_and_outcoming_link_ids(self, wb_polygon, model_part):
 
         for pump in f_pumps:
             # test if lines are crossing boundary of polygon
-            if pump.geometry().crosses(wb_polygon):
-                geom = pump.geometry().asPolyline()
+            pump_geometry = pump.geometry()
+            if pump_geometry.intersects(wb_polygon):
+                pump_end_node_id = pump["node_idx2"]
+                linestring = pump.geometry().asPolyline()
```
Comment (benvanbasten-ns) [inline tool_water_balance/tools/waterbalance.py:359]: Replace pump.geometry() with pump_geometry?

Comment (benvanbasten-ns) [inline tool_water_balance/tools/waterbalance.py]: Replace pump.geometry() with pump_geometry?

**Comment by benvanbasten-ns** [comment]
Added two minor comments

File: tool_water_balance/tools/waterbalance.py (hunk 1 above)
Comment (ldebek) [reply tool_water_balance/tools/waterbalance.py]: Done

File: tool_water_balance/tools/waterbalance.py (hunk 2 above)
Comment (ldebek) [reply tool_water_balance/tools/waterbalance.py]: Done.

**Pull request merged**